### PR TITLE
[CSM] Add function and chat command to disconnect from server.

### PR DESCRIPTION
**Pull request by red-001**
Part of #5394

### CODE REVIEW

File: doc/client_lua_api.md [hunk 1]
```diff
@@ -698,6 +698,8 @@ Call these functions only at load time!
 ### Client Environment
 * `minetest.get_player_names()`
     * Returns list of player names on server
+* `minetest.disconnect()`
+    * Disconnect from the server and exit to main menu.
```
Comment (nerzhul) [inline doc/client_lua_api.md]: missing return doc

File: src/client.cpp [hunk 2]
```diff
@@ -361,14 +363,12 @@ void Client::Stop()
 
 bool Client::isShutdown()
 {
-
-	if (!m_mesh_update_thread.isRunning()) return true;
```
Comment (nerzhul) [inline src/client.cpp]: are you sure it's good ?

**Comment by red-001** [comment]
Updated

**Comment by nerzhul** [comment]
Please fix codestyle of l_client.h https://travis-ci.org/minetest/minetest/jobs/217068742#L3411 and remove l_client.h from util/travis/clang-format-whiletest file please ?

File: src/script/lua_api/l_client.cpp [hunk 3]
```diff
@@ -107,6 +110,20 @@ int ModApiClient::l_send_respawn(lua_State *L)
 	return 0;
 }
 
+// disconnect()
+int ModApiClient::l_disconnect(lua_State *L)
+{
+	// Stops badly written Lua code form causing boot loops
+	if(getClient(L)->isShutdown()) {
```
Comment (nerzhul) [inline src/script/lua_api/l_client.cpp]: Codestyle https://travis-ci.org/minetest/minetest/jobs/217068742#L3386

**Comment by red-001** [comment]
Done

**Comment by nerzhul** [comment]
Now l_client.h is not happy with your code, please fix it. See https://travis-ci.org/minetest/minetest/jobs/217342356#L3160 This will make travis happy

**Comment by rubenwardy** [comment]
:+1: although untested

todo: minetest.register_on_shutdown though

**Pull request merged**